Support configuration as script arguments

hammady opened the pull request:
- Support configuration as script arguments, for example `--lat 20.23 --lng 50.552 --method Egypt`. These settings get persisted automatically so that later runs do not have to receive any arguments
- Set the volume of Azaan using arguments, for example: `--fajr-azaan-volume 1200` and `--azaan-volume 1800`
- Make the script portable by detecting the current directory
- Remove files ignored already like *.pyc and adhan.log

hammady committed:
Remove ignored files

hammady committed:
Make paths portable by detecting current dir

hammady committed:
Make print python3 compatible plus some linting

hammady committed:
Get location and method from arguments and save them

hammady committed:
Update README with simplified configuration

achaudhry commented on updateAzaanTimers.py:
```diff
@@ -12,29 +16,48 @@
-strPlayAzaanMP3Command = 'omxplayer -o local /home/pi/adhan/Adhan-Makkah.mp3 > /dev/null 2>&1'
-strUpdateCommand = 'python /home/pi/adhan/updateAzaanTimers.py >> /home/pi/adhan/adhan.log 2>&1'
-strClearLogsCommand = 'truncate -s 0 /home/pi/adhan/adhan.log 2>&1'
+strPlayFajrAzaanMP3Command = 'omxplayer --vol 1200 -o local {}/Adhan-fajr.mp3 > /dev/null 2>&1'.format(root_dir)
```
How do you feel about removing the vol argument from this line and next? People can use their speaker volume instead?

hammady replied:
You are right. I will introduce them as optional arguments instead.

hammady replied:
Fixed in ac6ca08.

hammady replied:
Wait, I will update the README accordingly.

hammady replied:
Updated in 1a56395.

achaudhry commented on updateAzaanTimers.py:
```diff
@@ -12,29 +16,48 @@
+    # save values
+    with open(file_path, 'wt') as f:
+        f.write('{},{},{}'.format(lat or '', lng or '', method or ''))
+    return lat or None, lng or None, method or None
```
Subsequent runs are crashing since this returns lat and lng as a string. Cast to float before returning
```suggestion
    return float(lat) or None, float(lng) or None, method or None
```

achaudhry reviewed (changes requested):
Jazak Allah these are great enhancements. Thank you for taking the time to add these. 

I left a few suggestions. One is a bug fix and the rest are minor updates. If you can make those updates that would be great. I'll merge once done.

Thanks again!

hammady committed:
Fix extraneous characters in README.md

Co-authored-by: Abrar Chaudhry <2893677+achaudhry@users.noreply.github.com>

hammady committed:
Fix bug of location coordinates being treated as strings

Co-authored-by: Abrar Chaudhry <2893677+achaudhry@users.noreply.github.com>

hammady committed:
Add volume levels as script arguments

hammady committed:
Update README to describe volume arguments

hammady commented:
> 
> 
> Jazak Allah these are great enhancements. Thank you for taking the time to add these.
> 
> I left a few suggestions. One is a bug fix and the rest are minor updates. If you can make those updates that would be great. I'll merge once done.
> 
> Thanks again!

Jazana Wa'eyakom, I have addressed all of your comments. Please let me know if anything more is needed to merge this.

Zamy97 commented:
I don't know if this would be the right spot to ask but let me know if not I'll remove this!

I am just playing with the **praytimes.py** and **updateazantimer.py**  and instead of playing the audio on a pi, I am trying to run this in my local mac with the help of crontab but what is confusing me is how could I work around this command that is used for omxplayer to run the audio. `omxplayer -o local /home/pi/adhan/Adhan-fajr.mp3 > /dev/null 2>&1` How could I work around that in order to be able to able to play the Adhan on my local machine?

Also,  it seems as my jobs gets created on my local machine but the audio file never really plays. 

Any tips or suggestions would be appreciated.

hammady commented:
> 
> 
> I don't know if this would be the right spot to ask but let me know if not I'll remove this!
> 
> I am just playing with the **praytimes.py** and **updateazantimer.py** and instead of playing the audio on a pi, I am trying to run this in my local mac with the help of crontab but what is confusing me is how could I work around this command that is used for omxplayer to run the audio. `omxplayer -o local /home/pi/adhan/Adhan-fajr.mp3 > /dev/null 2>&1` How could I work around that in order to be able to able to play the Adhan on my local machine?
> 
> Also, it seems as my jobs gets created on my local machine but the audio file never really plays.
> 
> Any tips or suggestions would be appreciated.

@Zamy97 that's off topic, but my answer is that for macOS the preferred way to run timed jobs is [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/ScheduledJobs.html). Although cron is still supported, but it is not recommended, and I couldn't get it to run on my Mac.

achaudhry commented:
Agreed with @hammady. @Zamy97 see my response on the [issue](https://github.com/achaudhry/adhan/issues/8) you've created. 

@hammady I'll look at the changes and merge them in tomorrow InshaAllah! Been busy with Eid and all :) Thanks for making the updates!

hammady commented:
> 
> 
> Agreed with @hammady. @Zamy97 see my response on the [issue](https://github.com/achaudhry/adhan/issues/8) you've created.
> 
> @hammady I'll look at the changes and merge them in tomorrow InshaAllah! Been busy with Eid and all :) Thanks for making the updates!

@achaudhry let me know if you have questions so far to be able to merge.

achaudhry commented:
@hammady sorry man, been extremely busy over the last few days but I haven't forgotten about this I promise :) Will look shortly. Thanks

hammady commented:
> 
> 
> @hammady sorry man, been extremely busy over the last few days but I haven't forgotten about this I promise :) Will look shortly. Thanks

Thanks @achaudhry, now #10 :)
